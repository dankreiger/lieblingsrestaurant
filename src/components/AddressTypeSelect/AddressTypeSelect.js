import React from 'react';
import { func } from 'prop-types';
import { Input } from 'reactstrap';
import { AddressTypeSelectWrapper } from './AddressTypeSelect.styles';

const AddressTypeSelect = ({ updateSearchFilterType }) => {
  return (
    <AddressTypeSelectWrapper>
      <Input
        type="select"
        name="addressTypeFilter"
        id="addressTypeFilter"
        onChange={e => updateSearchFilterType(e.target.value)}
      >
        <option value="establishments">Establishments</option>
        <option value="">All Addresses</option>
      </Input>
    </AddressTypeSelectWrapper>
  );
};

AddressTypeSelect.propTypes = {
  updateSearchFilterType: func
};

export default AddressTypeSelect;
