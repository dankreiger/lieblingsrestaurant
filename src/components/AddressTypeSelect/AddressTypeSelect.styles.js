import styled from 'styled-components';
import { FormGroup, Row } from 'reactstrap';
import {
  ACTIVE_NAV_SHADOW,
  DEFAULT_TRANSITION_DURATION,
  NAV_SHADOW
} from 'constants/index';

export const MapMenuRow = styled(Row)``;
export const AddressTypeSelectWrapper = styled(FormGroup)`
  margin-bottom: 0 !important;
  height: 100%;
  select {
    height: 100%;
    box-shadow: ${NAV_SHADOW};
    transition: box-shadow ${DEFAULT_TRANSITION_DURATION}ms
    &:focus {
      box-shadow: ${ACTIVE_NAV_SHADOW}
    }
  }
`;
