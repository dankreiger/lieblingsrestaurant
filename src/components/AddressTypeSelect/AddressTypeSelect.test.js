import React from 'react';
import { shallow } from 'enzyme';
import AddressTypeSelect from './AddressTypeSelect';
import { FormGroup, Input } from 'reactstrap';
import { AddressTypeSelectWrapper } from './AddressTypeSelect.styles';

describe('AddressTypeSelect', () => {
  let addressTypeSelectComponent;

  beforeEach(() => {
    addressTypeSelectComponent = shallow(<AddressTypeSelect />);
  });

  describe('rendering', () => {
    test('renders as expected', () => {
      expect(addressTypeSelectComponent.length).toBe(1);
      expect(addressTypeSelectComponent).toMatchSnapshot();
    });
  });

  describe('structure', () => {
    test('renders 1 <FormGroup> component with a <Input> select menu component', () => {
      const formGroup = addressTypeSelectComponent.find(
        AddressTypeSelectWrapper
      );
      expect(formGroup.length).toBe(1);

      expect(formGroup.find(Input).length).toBe(1);
      expect(formGroup.find(Input).props().type).toBe('select');
      expect(formGroup.find(Input).props().name).toBe('addressTypeFilter');
    });
  });
});
