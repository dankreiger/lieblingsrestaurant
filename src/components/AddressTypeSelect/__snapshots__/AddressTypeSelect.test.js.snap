// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressTypeSelect rendering renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddressTypeSelect />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Input
        id="addressTypeFilter"
        name="addressTypeFilter"
        onChange={[Function]}
        type="select"
      >
        <option
          value="establishments"
        >
          Establishments
        </option>
        <option
          value=""
        >
          All Addresses
        </option>
      </Input>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <option
            value="establishments"
          >
            Establishments
          </option>,
          <option
            value=""
          >
            All Addresses
          </option>,
        ],
        "id": "addressTypeFilter",
        "name": "addressTypeFilter",
        "onChange": [Function],
        "type": "select",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Establishments",
            "value": "establishments",
          },
          "ref": null,
          "rendered": "Establishments",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "All Addresses",
            "value": "",
          },
          "ref": null,
          "rendered": "All Addresses",
          "type": "option",
        },
      ],
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": true,
        "rules": Array [
          "
  margin-bottom: 0 !important;
  height: 100%;
  select {
    height: 100%;
    box-shadow: ",
          "1px 1px 3px #333",
          ";
    transition: box-shadow ",
          "250",
          "ms
    &:focus {
      box-shadow: ",
          "0px 0px 1px #333",
          "
    }
  }
",
        ],
      },
      "displayName": "Styled(FormGroup)",
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Input
          id="addressTypeFilter"
          name="addressTypeFilter"
          onChange={[Function]}
          type="select"
        >
          <option
            value="establishments"
          >
            Establishments
          </option>
          <option
            value=""
          >
            All Addresses
          </option>
        </Input>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <option
              value="establishments"
            >
              Establishments
            </option>,
            <option
              value=""
            >
              All Addresses
            </option>,
          ],
          "id": "addressTypeFilter",
          "name": "addressTypeFilter",
          "onChange": [Function],
          "type": "select",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Establishments",
              "value": "establishments",
            },
            "ref": null,
            "rendered": "Establishments",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "All Addresses",
              "value": "",
            },
            "ref": null,
            "rendered": "All Addresses",
            "type": "option",
          },
        ],
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
  margin-bottom: 0 !important;
  height: 100%;
  select {
    height: 100%;
    box-shadow: ",
            "1px 1px 3px #333",
            ";
    transition: box-shadow ",
            "250",
            "ms
    &:focus {
      box-shadow: ",
            "0px 0px 1px #333",
            "
    }
  }
",
          ],
        },
        "displayName": "Styled(FormGroup)",
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
